### PR TITLE
PCIE-80 layers dev implement abstract ohc class and its concrete classes

### PR DESCRIPTION
**Pull request by Nazeeh007**
PCIE-80 Added the classes

### CODE REVIEW

File: PCIe 6.0 Packet Generator and Exerciser/Google Test/Google Test.vcxproj [hunk 1]
```diff
@@ -33,9 +33,11 @@
   <ImportGroup Label="PropertySheets" />
   <PropertyGroup Label="UserMacros" />
   <ItemGroup>
+    <ClInclude Include="..\PCIe 6.0 Packet Generator and Exerciser\layers\globals.h" />
```
Comment (mohanadismail) [inline PCIe 6.0 Packet Generator and Exerciser/Google Test/Google Test.vcxproj:36]: Why are the globals files added to the Google Test project? Please remove them

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
+// Abstract class OHC
+class OHC {
```
Comment (mohanadismail) [inline PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h:2]: Make this class abstract

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    int size = 4;
+};
+
+
+// Concrete class OHCA1 that inherits from OHC
+class OHCA1 : public OHC {
+public:
+    // Public attributes firstDWBE and lastDWBE, both arrays of integers with a length of 4
+    int firstDWBE[4];
+    int lastDWBE[4];
```
Comment (mohanadismail) [inline PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h:14]: Use of bitset is more efficient

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+};
+
+// Concrete class OHCA3 that inherits from OHC
+class OHCA3 : public OHC {
+public:
+    // Public attributes firstDWBE and lastDWBE, both arrays of integers with a length of 4
+    int firstDWBE[4];
+    int lastDWBE[4];
```
Comment (mohanadismail) [inline PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h:31]: Use of bitset is more efficient

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    }
+};
+
+// Concrete class OHCA5 that inherits from OHC
+class OHCA5 : public OHC {
+public:
+    int destinationSegment;
+    int completerSegment;
+    int lowerAddress[2];
```
Comment (mohanadismail) [inline PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h:64]: Use of bitset is more efficient

File: PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    enum class CPLStatus {
+        True = 1,
+        False = 0
+    };
+    CPLStatus CPLStatus_;
```
Comment (mohanadismail) [inline PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/ohc.h:69]: Remove the underscore as this is not private

**Review by mohanadismail — CHANGES_REQUESTED** [review]

File: PCIe 6.0 Packet Generator and Exerciser/Google Test/Google Test.vcxproj (hunk 1 above)
Comment (Nazeeh007) [reply PCIe 6.0 Packet Generator and Exerciser/Google Test/Google Test.vcxproj:36]: done

**Review by mohanadismail — APPROVED** [review]

**Pull request merged**